### PR TITLE
updating asm_secrets_arns

### PR DESCRIPTION
**Pull request by sshaikRC**
### Submitter Checklist
-  [ ] Is there an impact on Auditing and Logging procedures or capabilities?
-  [ ] Is there an impact on Authentication procedures or capabilities?
-  [ ] Is there an impact on Authorization procedures or capabilities?
-  [ ] Is there an impact on Communication Security procedures or capabilities?
-  [ ] Is there an impact on Cryptography procedures or capabilities?
-  [ ] Is there an impact on Sensitive Data procedures or capabilities?
-  [ ] Is there an impact on any other security-related procedures or capabilities?
-  [ ] No security impacts identified.

Security Risks Identified - For any applicable items on the "Submitter Checklist," describe the impact of the change and any implemented mitigations.

### CODE REVIEW

**Comment by robo-gotham** [comment]
Snyk Scanning for Commit: b8acbfb67aedb6f913edccfc35700eebb8f6fe6d

Snyk Infrastructure as Code

- Snyk testing Infrastructure as Code configuration issues.
✔ Test completed.

Issues
  No vulnerable paths were found!

-------------------------------------------------------

</details>

Test Summary

  Organization: batcave-ispg
  Project name: CMS-Enterprise/batcave-tf-eso

✔ Files without issues: 4
✗ Files with issues: 0
  Ignored issues: 0
  Total issues: 0 [ 0 critical, 0 high, 0 medium, 0 low ]

-------------------------------------------------------

Report Complete

  Your test results are available at: https://snyk.io/org/batcave-ispg/projects
  under the name: CMS-Enterprise/batcave-tf-eso

File: iam.tf [hunk 1]
```diff
@@ -14,6 +14,18 @@ module "external_secrets_irsa" {
...
+    secret_names = [
+        "private-registry",
+        "batcave/argocd-config-batcave-dev",
+        "batcave/grafana-secret-batcave-dev",
+        "batcave/sso-secret-batcave-dev",
+        "batcave/istio-secret-batcave-dev",
+        "batcave/sonar-registry-credentials-batcave-dev",
+        "batcave/sonar-agent-api-key-batcave-dev",
+        "batcave/kiali-batcave-dev",
+        "batcave/gitlab-rails-secret-s3-batcave-dev"
```
Comment (bushong1) [inline iam.tf]: Remove "batcave-dev" suffix
```suggestion
        "batcave/argocd-config",
        "batcave/grafana-secret",
        "batcave/sso-secret",
        "batcave/istio-secret",
        "batcave/sonar-registry-credentials",
        "batcave/sonar-agent-api-key",
        "batcave/kiali",
        "batcave/gitlab-rails-secret-s3"
```

**Review by spantacms — APPROVED** [review]
Shanawaze ran the terragrunt and found no error on his implementation. Approving this PR

File: iam.tf (hunk 1 above)
Comment (sshaikRC) [reply iam.tf]: So the way i have the migration script now is it checks the aws profile and if its batcave-dev, it adds batcave-dev to the end of the secret name. If not it adds the cluster name to the end. Are you saying to just remove the suffix for the dev env?

**Review by spantacms — CHANGES_REQUESTED** [review]
Charles comment regarding batcave-dev. Please confirm with him and I will approve again

File: iam.tf (hunk 1 above)
Comment (bushong1) [reply iam.tf]: yeah, so that way these names can be consistent across environments.

**Review by spantacms — APPROVED** [review]

**Comment by spantacms** [comment]
As per Shanawaz Charles issue is resolve. I approved this change

File: iam.tf [hunk 2]
```diff
@@ -14,6 +14,18 @@ module "external_secrets_irsa" {
   }
 }
 
-data "aws_secretsmanager_secret" "batcave-private-registry" {
-  name = "batcave/registry-artifactory.cloud.cms.gov"
+locals {
+    secret_arns = [ for name in local.secret_names : "arn:aws:secretsmanager:${var.aws_region}:${var.aws_id}:secret:${name}"
+    ]
```
Comment (bushong1) [inline iam.tf]:
```suggestion
    secret_arns = [ for name in local.secret_names : "arn:aws:secretsmanager:${var.aws_region}:${var.aws_id}:secret:${name}" ]
```

**Review by bushong1 — APPROVED** [review]

File: iam.tf [hunk 3]
```diff
@@ -14,6 +14,18 @@ module "external_secrets_irsa" {
...
+    secret_names = [
+        "private-registry",
+        "batcave/argocd-config",
+        "batcave/grafana-secret",
+        "batcave/sso-secret",
+        "batcave/istio-secret",
+        "batcave/sonar-registry-credentials",
+        "batcave/sonar-agent-api-key",
+        "batcave/kiali",
+        "batcave/gitlab-rails-secret-s3"
```
Comment (bushong1) [inline iam.tf]:
```suggestion
        "batcave/gitlab-rails-secret-s3",
```

**Review by bushong1 — APPROVED** [review]
made 2 small formatting tweaks. looks good

**Pull request merged**